### DISCUSSION TITLE
PerUser instead of PerMachine bundle installation

### DISCUSSION THREAD

**Discussion by laurentcau** [wixtoolset/issues#7212] [Questions]
I have defined a bundle with several items in the Chain,
all items are marked ForcePerMachine="yes" for MsiPackage and PerMachine="yes" for ExePackage
but my bundle seems to be always installed in a PerUser mode.
How do I make it install as PerMachine ? In installation log I see:
```
    Line 148: [A164:C924][2023-02-01T13:52:09]w322: Skipping cross-scope dependency registration on package: eonUninstaller, bundle scope: PerUser, package scope: PerMachine
    Line 156: [A164:C924][2023-02-01T13:52:09]w322: Skipping cross-scope dependency registration on package: LumenRTAppInstaller, bundle scope: PerUser, package scope: PerMachine
    Line 159: [A164:C924][2023-02-01T13:52:09]w322: Skipping cross-scope dependency registration on package: LumenRT_Content1, bundle scope: PerUser, package scope: PerMachine
    Line 160: [A164:C924][2023-02-01T13:52:09]w322: Skipping cross-scope dependency registration on package: LumenRT_Content2, bundle scope: PerUser, package scope: PerMachine
    Line 161: [A164:C924][2023-02-01T13:52:09]w322: Skipping cross-scope dependency registration on package: LumenRT_Content3, bundle scope: PerUser, package scope: PerMachine
    Line 162: [A164:C924][2023-02-01T13:52:09]w322: Skipping cross-scope dependency registration on package: LumenRT_Content4, bundle scope: PerUser, package scope: PerMachine
   ```
   
What I understood is: if all packages are set PerMachine, Bundle should also be set PerMachine. So I make sure my MSI were set as PerMachine both in bundle xml and msi xml with InstallScope="perMachine".

**Comment by rseanhall** [comment]
If you run `wix burn extract path\to\bundle.exe -out path\to\extractfolder` you can look at `path\to\extractfolder\BA\manifest.xml` to see if there are any per-user packages.

**Reply by laurentcau** [reply]
I'm using v3.11, so I used dark.exe (found this tools name through stackoverflow).
I checked extractfolder\UX\manifest.xml, I don't see any PerUser packages.
The main installer seems to be set PerMachine too:
	<Registration Id="{xxxxxxxx}" ExecutableName="Setup_LumenRTx64_16.17.58.056.exe" **PerMachine="yes"** Tag="BE;SP=0;LCID=1033;Win64=True;pid=2517" Version="16.17.58.56" ProviderKey="{xxxxxx}">

**Reply by rseanhall** [reply]
v4 is getting very close to release. You should try it out to see if it is also being per-user when you don't expect it to be so that we could have a chance to fix it. This feature generally works so we would need more information to be able to reproduce this ourselves.

**Reply by laurentcau** [reply]
We should release the product very soon, so we don't plane to use v4.
Is there a safe place I can share you the bundle or can you share me your email so I can send you a link for download ?

**Comment by barnson** [comment]
You can install HeatWave and convert your authoring to test it. HeatWave and WiX v4 live side-by-side with WiX v3.

**Reply by laurentcau** [reply]
Our build pipeline is complex (based on sort of custom makefile), so even there are powerful visual studio plugins, it's not easy for us to integrate.
